gson/src/main/java/com/google/gson/internal/Excluder.java

```diff
@@ -168,11 +172,7 @@ public boolean excludeField(Field field, boolean serialize) {
       }
     }
 
-    if (!serializeInnerClasses && isInnerClass(field.getType())) {
-      return true;
-    }
-
-    if (isAnonymousOrNonStaticLocal(field.getType())) {
+    if (excludeClass(field.getType(), serialize)) {
```

Comment:
Changed this to avoid code duplication. This should be identical to the previous behavior, but previously `excludeClass` was called separately:
https://github.com/google/gson/blob/ff401f1545e71672e7090cbeab60e776cda0ac4d/gson/src/main/java/com/google/gson/internal/bind/ReflectiveTypeAdapterFactory.java#L80-L82

(and it looks like there are no other callers of `excludeField`)